src/main/java/net/coderbot/iris/mixin/fabulous/MixinDisableFabulousGraphics.java

```diff
@@ -34,12 +37,22 @@
 
 		if (!Iris.getIrisConfig().areShadersEnabled()) {
 			// Nothing to do here, shaders are disabled.
+			if (this.wasFabulous) {
+				options.graphicsMode = GraphicsStatus.FABULOUS;
+				this.wasFabulous = false;
+			}
 			return;
 		}
 
 		if (options.graphicsMode == GraphicsStatus.FABULOUS) {
 			// Disable fabulous graphics when shaders are enabled.
 			options.graphicsMode = GraphicsStatus.FANCY;
+			this.wasFabulous = true;
```

Comment:
Won't this cause wasFabulous to get set to `true` if the user is cycling from Fancy to Fast  in video settings?